src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/detection_output/detection_output_kernel_ref.cpp

```diff
@@ -246,17 +246,22 @@ KernelsData DetectionOutputKernelRef::GetKernelsData(const Params& params, const
                                        MakeJitConstant("NUM_PRIOR_BLOCKS", num_score_block)});
             }
         } else if (i == 1) {
-             if (detectOutParams.detectOutParams.decrease_label_id) {
+            if (detectOutParams.detectOutParams.decrease_label_id) {
+                // Always use local memory since LWS size is 1x1x16 (16 WI * 100 (stack size) * 4 (int size) = 6.25 KB of SLM memory)
+                cldnnJit.AddConstant(MakeJitConstant("USE_LOCAL_MEMORY_FOR_STACK", "true"));
                 cldnnJit.AddConstants({MakeJitConstant("DO_STAGE_" + std::to_string(i) + "_MXNET", "true"),
                                        MakeJitConstant("LOCAL_WORK_NUM", dispatchData.lws[2]),
                                        MakeJitConstant("PARTITION_STEP", GetPartitionStep(dispatchData.lws[2]))});
-             } else {
+            } else {
+                // Limit local memory usage by 8*16 LWS (128 WI * 100 (stack size) * 4 (int size) = 50 KB of SLM memory)
+                if (dispatchData.lws[0] * dispatchData.lws[1] <= 8 * 16)
+                    cldnnJit.AddConstant(MakeJitConstant("USE_LOCAL_MEMORY_FOR_STACK", "true"));
```

Comment:
shouldnt this flag be separated for each stage? e.g., what if stage 1 can use local mem for stack but stage2 cant?

Reply:
Hm, as I can see, all of the "initial" macros will be generated for each processing stage as if they are different kernels with the single difference: corresponding DO_STAGE_# will be set to true. If so, stage1's  USE_LOCAL_MEMORY_FOR_STACK will not affect stage3, because it will not be set to true during code generation for stage3. Isn't it?

Reply:
Ah yes thats true.